src/utils/validator/AppointmentInputValidator.js

```diff
@@ -0,0 +1,66 @@
+import { UserInputError } from "apollo-server-express";
+
+import { User, Service, Branch } from "../../models";
+
+/**
+ * Validate the appointment to create, considering fields validity of:
+ * - appointmentdate
+ * - customerID
+ * - branchID
+ * - serviceID
+ * - vehicleID
+ * - serviceID
+ * @param {Object} appointmentInput
+ * @param {Object} user
+ */
+export const appointmentInputValidator = async (appointmentInput, user) => {
+  // validate the appointment date
+  const validDate = new Date(appointmentInput.appointmentDate) > Date.now();
+
+  // validate current customer creates an appointment
+  const validCustomer = creatorValidator(appointmentInput.customerID, user.id);
```

Comment:
if employee or branch staff want to react to the appointment how to update?

Reply:
ok I will update